src/SquidConfig.h

```diff
@@ -64,7 +64,7 @@ class DiskConfig {
     ~DiskConfig() { delete[] swapDirs; }
 
     RefCount<SwapDir> *swapDirs = nullptr;
-    int n_allocated = 0;
+    unsigned int n_allocated = 0;
     int n_configured = 0;
     /// number of disk processes required to support all cache_dirs
     int n_strands = 0;
```

Comment:
I agree that `n_allocated` type should be unsigned, but we should not make types even less consistent and force type casts to achieve that "unsigned n_allocated" goal because those things make the code even worse overall.

Fixing such problems is often difficult because they tend to snowball (and this is one of those places). No recipe can work in all cases. Often, the following approach helps:

1. Focus on one class at a time (e.g., DiskConfig)
2. Change all the related fields at once (e.g., all `DiskConfig::n_*` data members). Repeated types usually need names, with a few exceptions (e.g., use `size_t` for single-object capacity/counting, like STL does and use `uint64_t` for response content lengths and other things that may exceed a single memory buffer in size)
3. Adjust the rest of the code as needed (e.g., to make it compile) while avoiding C type casts and other "regressions". Sometimes, more variables or data members need to be changed. Sometimes, an occasional static_cast or similar is a good idea to stop the snowball from growing.

If you find changing a lot of similar code (e.g., all the Config.cacheSwap.n_configured iteration loops), then it may be time to pause and refactor `Config.cacheSwap.swapDirs` storage to use an STL container and C++ for-range loops. That is an even bigger change, but it is long overdue and it will change a lot of the same code.

If the changes get too difficult or likely to get too large, please stop and discuss. It is much easier, on many levels, to move forward _before_ a lot of problematic changes have been committed!

Reply:
Thank you for the prompt and considerate reply.

I confess I didn't have much time to come up with a better fix, so I apologize.  I understand your rationale and agree with it; my first attempt was to change `n_allocated` et al to `size_t`, but that requires a lot more changes in other places and my main goal was to fix the FTBFS.

Anyway, I will go back to the drawing board and work on a better solution.  This might take some time because I'm very busy with other stuff (we're entering the final stages before the next Ubuntu is released).  Thanks again.

Reply:
Glad we are on the same page! I will mark this PR as a Draft to reflect that you are looking for a better solution (and to slightly reduce CI costs).

There is no rush, of course. And if a quick workaround is needed, we can probably cast/hack something where the allocation is happening without changing the underlying types around the code. Again, please let me know if you need my help with that.

Reply:
This is resolved by [2d232fa](https://github.com/squid-cache/squid/pull/1118/commits/2d232fadbbcc6c65961b6313770cfee113da94ab)